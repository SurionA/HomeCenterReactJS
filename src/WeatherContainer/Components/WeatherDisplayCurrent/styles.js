export const iconStyle = {
  fontSize: '10rem',
  margin: '35px 0',
};

export const temperatureStyle = {
  fontSize: '2rem',
};

export default {
  iconStyle,
  temperatureStyle,
};
