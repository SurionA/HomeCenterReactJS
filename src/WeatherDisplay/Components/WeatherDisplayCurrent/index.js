import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Row } from "react-bootstrap";
import { iconStyle, temperatureStyle } from "./styles";

function WeatherDisplayCurrent({ weatherData }) {
  return (
    <Fragment>
      <Row>
        <h3 className="title">{weatherData.name}</h3>
      </Row>
      <Row>
        <div style={iconStyle} className={`wi wi-icon-${weatherData.icon}`} />
      </Row>
      <Row style={temperatureStyle}>
        {Math.round(parseInt(weatherData.temperature, 10))}Â°
      </Row>
    </Fragment>
  );
}

WeatherDisplayCurrent.propTypes = {
  weatherData: PropTypes.shape({
    name: PropTypes.string,
    humidity: PropTypes.number,
    temperature: PropTypes.number,
    icon: PropTypes.number
  })
};

WeatherDisplayCurrent.defaultProps = {
  weatherData: PropTypes.shape({
    name: "DefaultName"
  })
};
export default WeatherDisplayCurrent;
