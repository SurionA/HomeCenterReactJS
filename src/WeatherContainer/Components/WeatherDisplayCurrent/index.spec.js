import React from 'react';
import renderer from 'react-test-renderer';
import WeatherDisplayCurrent from './';

it('renders correctly WeatherDisplayCurrent component', () => {
  const weatherData = {
    name: 'City Name',
    humidity: 65,
    temperature: 15,
    icon: 800,
  };
  const tree = renderer.create(<WeatherDisplayCurrent weatherData={weatherData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
