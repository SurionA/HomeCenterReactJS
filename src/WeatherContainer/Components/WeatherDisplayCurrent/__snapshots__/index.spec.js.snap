// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly WeatherDisplayCurrent component 1`] = `
Array [
  <div
    className="row"
>
    <h3
        className="title"
    >
        City Name
    </h3>
</div>,
  <div
    className="row"
>
    <div
        className="wi wi-icon-800"
        style={
            Object {
                "fontSize": "10rem",
                "margin": "35px 0",
              }
        }
    />
</div>,
  <div
    className="row"
    style={
        Object {
            "fontSize": "2rem",
          }
    }
>
    15
    Â°
</div>,
]
`;
